internal/attestation/aws/snp/validator.go

```diff
@@ -65,9 +73,17 @@ func getTrustedKey(_ context.Context, attDoc vtpm.AttestationDocument, _ []byte)
 	return pubArea.Key()
 }
 
-// validateSNPReport validates the report by checking if it has a valid VLEK signature.
+// Validate a given SNP report.
+type snpReportValidator interface {
+	validate(attestation vtpm.AttestationDocument, kdsClient askGetter, ark *x509.Certificate, userData []byte) error
+}
+
+type awsValidator struct{}
+
+// validate the report by checking if it has a valid VLEK signature.
 // The certificate chain ARK -> ASK -> VLEK is also validated.
-func (v *Validator) validateSNPReport(attestation vtpm.AttestationDocument, _ *attest.MachineState) error {
+// Checks that the report's userData matches the connection's userData.
+func (awsValidator) validate(attestation vtpm.AttestationDocument, kdsClient askGetter, ark *x509.Certificate, userData []byte) error {
```

Comment:
Since this function is called from `getTrustedKey`, which takes a context as argument, we can hand it down to this function and avoid creating a `context.Background()` on the fly